src/UOrders.Api/Services/OrdersRepository.cs

```diff
@@ -81,7 +81,7 @@ public decimal GetOrderTotalPrice(int orderId)
             where !price.ValidTo.HasValue || price.ValidTo > dbOrder.OrderedOn
             select price.Value * item.Count;
 
-        return basePriceComponents.Union(optionPriceComponents).Sum();
```

Comment:
the union function removes any duplicate elements. But in order to calculate the price correctly we need to respect these. Summing these collections individually and adding the results together fixes the issue.